documentation/dataplane/dash-flow-api.md

```diff
@@ -0,0 +1,534 @@
+| SAI_FLOW_ENTRY_ATTR_FLOW_DATA_PB    | `sai_u8_list_t` | The flow data protocol buffer enables high-efficiency creation, retrieval, and communication for a flow entry. |
+
+### Flow Bulk Get Session
+
+To manage data transfer to a server via gRPC, we introduce a flow entry bulk session that incorporates filtering capabilities to precisely define the data range for transfer. The procedure for setting up these filters is straightforward:
+
+1. Initially, create up to five flow bulk get session filters based on the specific needs for filtering the flows.
+2. Subsequently, establish a flow bulk get session filter and integrate these filters as attributes.
+
+For example, consider the scenario where it's necessary to select all flow entries with a version less than 5 and greater than version 3. In this case, two distinct filters should be defined to meet the criteria and then create the flow entry bulk session.
+
+The filter, defined as an object, is specified as follows:
+
+| Function                                         | Description                                                  |
+| ------------------------------------------------ | ------------------------------------------------------------ |
+| create_flow_entry_bulk_get_session_filter        | Add a single new filter for flow entry bulk get session feature. |
+| remove_flow_entry_bulk_get_session_filter        | Remove a single filter for flow entry bulk get session feature. |
+| set_flow_entry_bulk_get_session_filter_attribute | Set attributes for a single filter in flow entry bulk get session. |
+| get_flow_entry_bulk_get_session_filter_attribute | Get attributes of a single filter in flow entry bulk get session. |
+| create_flow_entry_bulk_get_session_filters       | Add multiple new filters for flow entry bulk get session feature. |
+| remove_flow_entry_bulk_get_session_filters       | Remove multiple filters for flow entry bulk get session feature. |
+
+| Attribute Name                                               | Type                                                | Description                                                 |
+| ------------------------------------------------------------ | --------------------------------------------------- | ----------------------------------------------------------- |
+| SAI_FLOW_ENTRY_BULK_GET_SESSION_FILTER_ATTR_DASH_FLOW_ENTRY_BULK_GET_SESSION_FILTER_KEY | `sai_dash_flow_entry_bulk_get_session_filter_key_t` | Key of the filter                                           |
+| SAI_FLOW_ENTRY_BULK_GET_SESSION_FILTER_ATTR_DASH_FLOW_ENTRY_BULK_GET_SESSION_OP_KEY | `sai_dash_flow_entry_bulk_get_session_op_key_t`     | Operation of the filter                                     |
+| SAI_FLOW_ENTRY_BULK_GET_SESSION_FILTER_ATTR_INT_VALUE        | `sai_uint64_t`                                      | INT Value of the filter (Mutually exclusive with IP Value.) |
```

Comment:
if you like to restrict the condition, feel free to check out the `@validonly` tag in SAI.